modules/ting/src/Search/TingSearchFacet.php

```diff
@@ -34,16 +40,30 @@ class TingSearchFacet {
    *
    * @param string $name
    *   Name of the facet.
-   *
    * @param TingSearchFacetTerm[] $terms
    *   Optional list of terms.
    */
-  public function __construct($name, $terms = []) {
+  public function __construct($name, array $terms = []) {
     $this->name = $name;
     // Set via setter to store the terms by their names.
     $this->setTerms($terms);
   }
 
+  /**
+   * Get the facet type.
+   *
+   * @return string
+   *   The type.
+   */
+  public function getType() {
+    switch ($this->getName()) {
+      case 'facet.date':
+        return self::TYPE_INTERVAL;
+    }
+
+    return self::TYPE_DEFAULT;
```

Comment:
This ties the facets to OpenSearch naming. That is unfortunate. Instead you should pass the type in the constructor just as you do with the name and the terms.